test/skate/repo_test.exs

```diff
@@ -0,0 +1,48 @@
+defmodule Skate.RepoTest do
+  use ExUnit.Case, async: false
+
+  describe "add_prod_credentials/2" do
+    setup do
+      original_hostname = System.get_env("POSTGRES_HOSTNAME", "localhost")
+      original_port = System.get_env("POSTGRES_PORT", "5432")
+      original_username = System.get_env("POSTGRES_USERNAME", System.get_env("USER"))
+
+      System.put_env("POSTGRES_HOSTNAME", "db_server_hostname")
+      System.put_env("POSTGRES_USERNAME", "my_username")
+      System.put_env("POSTGRES_PORT", "6789")
+
+      on_exit(fn ->
+        System.put_env("POSTGRES_HOSTNAME", original_hostname)
+        System.put_env("POSTGRES_USERNAME", original_username)
+        System.put_env("POSTGRES_PORT", original_port)
```

Comment:
If the defaults above get used, this doesn't perfectly restore the environment. You'd have to detect that it was unset and call `System.delete_env()` instead.